src/types/excelreader.rs

```diff
@@ -29,17 +64,35 @@ impl ExcelReader {
             .with_context(|| format!("Could not open workbook at {path}"))?;
         let sheet_names = sheets.sheet_names().to_owned();
         Ok(Self {
-            sheets,
+            sheets: ExcelSheets::File(sheets),
+            sheet_names,
+            source: path.to_owned(),
+        })
+    }
+}
+
+impl TryFrom<&[u8]> for ExcelReader {
+    type Error = FastExcelError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        let bytes: &'static [u8] = Box::leak(bytes.into());
```

Comment:
IMO it's fine to do that to avoid the need to collect into a vec or something but WDYT @lukapeschke?

Reply:
Hmm I think this needs some tests because from what I understand from the PyO3 docs, it's not clear wether `extract::<>()`ing a Python object clones the data or not. Because if it does, I think `leak` will indeed memory leak here since the memory won't be freed by Python. Could you do a comparison with an `extract::<Vec<u8>>` rather than `&[u8]` ? If the perf/memory consumption is the same, it would probably be easier to work on owned types directly, wdyt ?

Reply:
Yes you're right I need to dig a bit